### PR TITLE
Create new validations for admin

### LINKED ISSUE #22: Implement code validations for contract functions
## Context 
We need to implement a way to ensure that only the admin of the contract is authorized to execute operations within the contract, it can be a function which validates this with the invoker that can be executed on the contract functions that require it or find a better way for doing this.

### PR DESCRIPTION
**Pull request by Jenietoc**
## Description

Added validations at the moment of adding and deleting members.

Fixes #22 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings

### CODE REVIEW

File: mtk-automation/src/contract_actions/member.rs [hunk 1]
```diff
@@ -3,16 +3,20 @@
 //! This module lets you get and revoke members of the organization
 use crate::contract_actions::datakey::DataKey;
 use crate::contract_actions::token_operation;
+use crate::contract_actions::admin;
 use soroban_sdk::{vec, Address, Env, RawVal, Vec};
 
-pub(crate) fn add_member(env: &Env, account: &Address) {
+pub(crate) fn add_member(env: &Env, member: Address, admin: Address) {
+    admin::check_admin(env, &admin);
+    if is_member(env, &member) {panic!("Member Already exists")};
```
Comment (Odraxs) [inline mtk-automation/src/contract_actions/member.rs:11]: @Jenietoc  Is this code well formatted?
pls run `cargo fmt` to ensure it is well formatted
@L-Zuluaga  I think we need to add `cargo fmt --check` to the CI file

**Review by Odraxs — CHANGES_REQUESTED** [review]

File: mtk-automation/src/contract_actions/member.rs [hunk 2]
```diff
@@ -1,18 +1,24 @@
 //! Module Member
 //!
 //! This module lets you get and revoke members of the organization
+use crate::contract_actions::admin;
 use crate::contract_actions::datakey::DataKey;
 use crate::contract_actions::token_operation;
 use soroban_sdk::{vec, Address, Env, RawVal, Vec};
 
-pub(crate) fn add_member(env: &Env, account: &Address) {
+pub(crate) fn add_member(env: &Env, member: Address, admin: Address) {
```
Comment (L-Zuluaga) [inline mtk-automation/src/contract_actions/member.rs:9]: You are receiving the admin as an address, but we could use the env.invoker() function. However, that function was removed on the latest version, so I think we should use the new approach, which required to include the address as a param WITH some additional steps.

Take a look at this and check if it make sense: https://soroban.stellar.org/docs/how-to-guides/auth-migration#migrating-envinvoker
image

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]

File: mtk-automation/src/contract_actions/admin.rs [hunk 3]
```diff
@@ -13,3 +13,14 @@ pub(crate) fn get_admin_id(env: &Env) -> Address {
...
 }
+
+pub fn has_administrator(e: &Env) -> bool {
+    let key = DataKey::AdminId;
+    e.storage().has(&key)
+}
+
+pub fn check_admin(e: &Env, admin: &Address) {
+    if admin != &get_admin_id(e) {
+        panic!("Not authorized by admin")
```
[the remark this answers is not in the document]
Comment (Jenietoc) [reply mtk-automation/src/contract_actions/admin.rs:24]: **Not Authorized** looking good

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -37,4 +37,4 @@ jobs:
           --wasm target/wasm32-unknown-unknown/release/mtk_automation.wasm \
           --secret-key ${{ secrets.DEPLOYER_SECRET }} \
           --rpc-url https://horizon-futurenet.stellar.cash:443/soroban/rpc \
-          --network-passphrase 'Test SDF Future Network ; October 2022'
+          --network-passphrase 'Test SDF Future Network ; October 2022'
```
Comment (L-Zuluaga) [inline .github/workflows/cd.yaml:40]: include here the EOL

**Review by L-Zuluaga — CHANGES_REQUESTED** [review]

**Review by L-Zuluaga — APPROVED** [review]
LGTM
@einerzg if you can give it a look just in case would be awesome!

**Review by einerzg — APPROVED** [review]

**Review by Odraxs — APPROVED** [review]

**Pull request merged**